### PR TITLE
DOCS-#3204: Add a section regarding general rules for committers

### LINKED ISSUE #3204: General rules for commiters
We should add a section defining general rules for committers to Contributing page.

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3204  
- [x] tests passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -105,10 +105,17 @@ because it links commits to their issues.
 
 The commit message should follow a colon (:) and be descriptive and succinct.
 
+General Rules for committers
+----------------------------
+
+- Try to keep a PR name as much descriptive as possible
+- Try to keep PRs as small as possible. One PR should be making one semantically atomic change
+- Don't merge own PRs even If you are eligible to do it
```
Comment (vnlitvinov) [inline docs/contributing.rst:113]:
```suggestion
- Don't merge your own PRs even if you are technically able to do it
```

**Review by vnlitvinov — DISMISSED** [review]
LGTM

File: docs/contributing.rst [hunk 2]
```diff
@@ -105,10 +105,17 @@ because it links commits to their issues.
 
 The commit message should follow a colon (:) and be descriptive and succinct.
 
+General Rules for committers
+----------------------------
+
+- Try to keep a PR name as much descriptive as possible
```
Comment (devin-petersohn) [inline docs/contributing.rst:111]:
```suggestion
- Try to write a PR name as descriptive as possible.
```

File: docs/contributing.rst [hunk 3]
```diff
@@ -105,10 +105,17 @@ because it links commits to their issues.
...
+- Try to keep PRs as small as possible. One PR should be making one semantically atomic change
```
Comment (devin-petersohn) [inline docs/contributing.rst:112]:
```suggestion
- Try to keep PRs as small as possible. One PR should be making one semantically atomic change.
```

File: docs/contributing.rst [hunk 4]
```diff
@@ -105,10 +105,17 @@ because it links commits to their issues.
...
+- Don't merge your own PRs even if you are technically able to do it
```
Comment (devin-petersohn) [inline docs/contributing.rst:113]:
```suggestion
- Don't merge your own PRs even if you are technically able to do it.
```

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
A few minor rewordings, thanks for putting this in @YarShev!

**Review by vnlitvinov — APPROVED** [review]
LGTM

**Comment by YarShev** [comment]
@devin-petersohn , reminder

**Review by devin-petersohn — APPROVED** [review]
Thanks @YarShev!

**Pull request merged**